packages/bsky/src/api/app/bsky/feed/getAuthorFeed.ts

```diff
@@ -82,7 +82,10 @@ export const skeleton = async (inputs: {
 
   const isFirstPageRequest = !params.cursor
   const shouldInsertPinnedPost =
-    isFirstPageRequest && params.includePins && !!actor.profile?.pinnedPost
+    isFirstPageRequest &&
+    params.includePins &&
+    !!actor.profile?.pinnedPost &&
+    actor.profile.pinnedPost.uri.includes(actor.did)
```

Comment:
Any chance of there being a handle in the pinned post URI?

Reply:
I don't believe that if it was a handle in there that it would get resolved during hydration, so it wouldn't be a valid pinned post URI anyway